### ISSUE TITLE
RUF055: false negative for `re.search(...)`

### ISSUE THREAD

**Issue by trim21** [astral-sh/ruff#14782]
```python
import re


def test_cost_repeated_merge(errors):
    assert any(re.search("Duplicate merge", error.message) for error in errors)

```

```text
$ ruff check d.py --select RUF055 --preview
All checks passed!

$ ruff --version
ruff 0.8.1
```

**Comment by MichaReiser** [comment]
The issue here is that Ruff doesn't detect that `re.search` is used in a boolean position.

**Comment by trim21** [comment]
> The issue here is that Ruff doesn't detect that `re.search` is used in a boolean position. I don't think it's worth special casing `any` (and many other functions that take boolean arguments)

Wouldn't it make more sence to rewrite `re.search("Duplicate merge", error.message)` to `"Duplicate merge" in error.message` to fit in boolean type?

**Comment by MichaReiser** [comment]
It's not about the rewrite. It's that Ruff doesn't think it's safe the change the `search` call because it doesn't know that the `re.search` result is used in a boolean position - the code only tests if the result is true or false.

**Comment by trim21** [comment]
OK， I misunderstand. I thought that ruff know it's used as bool and decide not to report a issue.
